unstructured/ingest/connector/weaviate.py

```diff
@@ -0,0 +1,152 @@
+import json
+import typing as t
+from dataclasses import dataclass
+
+from unstructured.ingest.error import DestinationConnectionError
+from unstructured.ingest.interfaces import (
+    BaseConnectorConfig,
+    BaseDestinationConnector,
+    BaseIngestDoc,
+    WriteConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.utils import requires_dependencies
+
+
+@dataclass
+class SimpleWeaviateConfig(BaseConnectorConfig):
+    host_url: str
+    class_name: str
+    auth_keys: t.Optional[t.Dict[str, str]] = None
+
+
+@dataclass
+class WeaviateWriteConfig(WriteConfig):
+    batch_size: int = 100
+
+
+@dataclass
+class WeaviateDestinationConnector(BaseDestinationConnector):
+    write_config: WeaviateWriteConfig
+    connector_config: SimpleWeaviateConfig
+
+    @requires_dependencies(["weaviate"], extras="weaviate")
+    @DestinationConnectionError.wrap
+    def initialize(self):
+        from weaviate import Client
+
+        auth = self._resolve_auth_method()
+        self.client: Client = Client(url=self.connector_config.host_url, auth_client_secret=auth)
+
+    def check_connection(self):
+        pass
```

Comment:
Can we implement this based on a previous comment? Should be fine to just create a `Client()` instance. 

If creating a `Client()` instance is not enough to establish a connection & truly validate our credentials, we can make a basic query like checking a state via the client instance, so that it actually forms a connection